### DISCUSSION TITLE
Tile 2x4 playcards of specific size into A4 format PDF

### DISCUSSION THREAD

**Discussion by kofaysi** [ImageMagick/ImageMagick#5679] [Help] [answered]
Dear users, I'm struggling to tile images of playcards and print them in actual size and I seek help and advice here.

I expect an A4 format PDF portrait orientation page with 2x2 tiles of the rotated cards. Detailed specifications follow:

* the format of the resulting file to print shall be A4, 210mmx297mm (print as is, do not fit)
* the input images (playcards) can have various sizes (one size per batch), but must fit into a standard frame of 98.3mmx76.2mm, do not change aspect ratio, shrinking and resizing is allowed
* Optimize the page layout, i.e, tile as many tiles per page as possible
* A card of an atypical batch of cards has dimensions 382 pixels * 240 pixels
* The cards shall be rotated and placed in a grid of 2 by 2 to allow for some white space in the sides 

Up to now I converged to command

`montage -rotate 90 card*.png -size 210x297  -tile 2x2x10x10 -geometry 98.3x76.2+5+5 -quality 05 cards.pdf `

A sample card, make a copy of four 
![card1](https://user-images.githubusercontent.com/17022173/196804996-b94509e7-ba71-42d7-8f1c-b81fe0c48dc1.png)

My results: [cards.pdf](https://github.com/ImageMagick/ImageMagick/files/9824398/cards.pdf)
I expected an A4 format PDF portrait orientation, with 2x2 tiles of the rotated cards, but the command fails to keep the page size. The PNGs are displayed with low resolution (bad quality) -- this is not visible with the example.

**Comment by snibgo** [comment]
> -size 210x297

Do you really want something to be 210 pixels wide and 297 pixels high? I don't think so.

What size are your cards, in pixels? How large do you want your A4 print to be, in pixels?

**Reply by kofaysi** [reply]
> What size are your cards, in pixels? How large do you want your A4 print to be, in pixels?

The current cards are 382 by 240 pixels. They shall show up as 98.3mmx76.2mm images on A4 print. Why should I care about the pixel number in A4? I only expect that the resolution will be decent for printing (150dpi or 300dpi, but I do not really care about the optimization of the file size now)

**Comment by kofaysi** [comment]
Thank you for the hint.

After more tens of iterations I ended up with 
`montage card[1-4]*.png -tile 2x2+140+140 -geometry 382x240+65+52 -density 100 cards.pdf`
to achieve the desired tile placing and dimensions of the tiles. But I have still no clue which geometry of which option does what trick. I imagine that I have cards with different dimensions C and D and I do not know how to change the command. 
[cardsB.pdf](https://github.com/ImageMagick/ImageMagick/files/9846543/cardsB.pdf)

I searched the internet for some reference card to imagemagick montage, but found nothing. Do you know _any_ visualization of the options to montage, please?

**Comment by snibgo** [comment, marked as answer]
> Why should I care about the pixel number in A4?

ImageMagick works in pixels, not millimetres. When you create an image, or crop or resize or whatever, you are working in pixels, and IM needs the dimensions in pixels. You can then set the density in dpi or dots per centimetre.

> Do you know any visualization of the options to montage, please?

https://imagemagick.org/script/montage.php
https://imagemagick.org/Usage/montage/

**Comment by kofaysi** [comment]
OK, so using a spreadsheet (LO Calc) I came up with formulas to clean up the results
`montage card*.png -tile 2x2 -geometry 585x400+60+50 -density 47.6 -units PixelsPerCentimeter cards.pdf`

[cards.pdf](https://github.com/ImageMagick/ImageMagick/files/9847501/cards.pdf)

I was thinking foolishly that I can come up with the numbers in the command line.
![image](https://user-images.githubusercontent.com/17022173/197417597-adb64e79-fc6a-4d3f-93ed-7e51cf3f58dd.png)
